### DISCUSSION TITLE
Select text based off of Text Color

### DISCUSSION THREAD

**Discussion by rynsp8** [UglyToad/PdfPig#658] [Q&A] [answered]
How could I select a line of text based off of color?  
We have multiple PDFs of release notes from a vendor and their paragraph headers are always a certain shade of blue.  I'm thinking there may be a way using `Structure` but I'm a novice and don't know much about PDF structures and specifications.

**Comment by rynsp8** [comment]
My first thought is to use something like `if(word.Text.[some property] == [some color])`  but again, I'm not sure how to access the formatting of the actual strings.

**Comment by BobLd** [comment]
@rynsp8 if you already have your words, you should be able to access the letters that compose them. The color information is contained at the letter level, not word, nor line (you could think of a word with letters of different colors)

**Reply by rynsp8** [reply]
BobLd, thank you for the input I will give that a shot.  Now, can I ask another question?  How do you know that color information is contained at the letter level?  I've been looking at the ISO documentation, which is pretty in-depth, but hadn't yet found any mention about color being contained at the letter level.

**Comment by BobLd** [comment]
@rynsp8 in a pdf document, the color is not at letter level itself - correct. The instructions contained in a pdf document are about drawing stuff. So you could have a command that is a 'paint command', and it will paint the current object.

But PdfPig is done in a way so that you don't need to read the ISO documentation, and helps you access a lot of elements/information contained in a pdf file. For example it provides you with the letter of the color, but also with the colors of paths (drawings). Another example is that a pdf document has no concept of 'word', 'line' or 'paragraph'. PdfPig does the post processing for you so that's easier for you

**Reply by rynsp8** [reply]
BobLd, Again thanks for working with me on this.

Potentially the last thing, trying to select for certain rgb value, would the following work? 
`foreach (var letter in letters)
{
    if(letter.Color == (r,g,b))
         console.writeline(letter)
}`

Is that feasible?

**Reply by BobLd** [reply, marked as answer]
@rynsp8 it should more less yes, I think you would newd to call `letter.Color.ToRgb()` to get the RGB value.

**Reply by rynsp8** [reply]
Hey Thank BobLd!!  Below is what I've written and it's working as needed.

The first Console.WriteLine that is commented was to grab the RGB values and the method was ToRGBValues.  Thanks again.

`
foreach (var letter in letters) 
                {
                    //Console.WriteLine("{0}\t\t{1}\t\t{2}", letter, letter.Color.ToRGBValues(), letter.FillColor);
                    if (letter.Color.ToRGBValues() == (0, 0.506, 0.639))
                    {
                        Console.WriteLine(letter);
                    }
                }
`
